File: src/platform/Platform.ts
import { bridge } from '../native/bridge'
import { native } from '../native/native'

@bridge('dezel.platform.Platform')

/**
 * @class Platform
 * @since 0.1.0
 */
export class Platform {

	//--------------------------------------------------------------------------
	// Static
	//--------------------------------------------------------------------------

	/**
	 * The current platform.
	 * @property current
	 * @since 0.7.0
	 */
	public static get current(): Platform {

		if (current == null) {
			current = new Platform()
		}

		return current
	}

	//--------------------------------------------------------------------------
	// Property
	//--------------------------------------------------------------------------

	/**
	 * The platform's name.
	 * @property name
	 * @since 0.1.0
	 */
	@native public readonly name!: string

	/**
	 * The platform's version.
	 * @property version
	 * @since 0.1.0
	 */
	@native public readonly version!: string

}

/**
 * @const current
 * @since 0.7.0
 * @hidden
 */
let current: Platform | null = null

File: platforms/android/Dezel/dezel/src/main/java/ca/logaritm/dezel/modules/platform/Platform.kt
package ca.logaritm.dezel.modules.platform

import android.os.Build
import ca.logaritm.dezel.core.JavaScriptClass
import ca.logaritm.dezel.core.JavaScriptContext
import ca.logaritm.dezel.core.JavaScriptGetterCallback

/**
 * @class Platform
 * @since 0.1.0
 * @hidden
 */
open class Platform(context: JavaScriptContext): JavaScriptClass(context) {

	//--------------------------------------------------------------------------
	// JS Functions
	//--------------------------------------------------------------------------

	/**
	 * @method jsGet_name
	 * @since 0.1.0
	 * @hidden
	 */
	open fun jsGet_name(callback: JavaScriptGetterCallback) {
		callback.returns("android")
	}

	/**
	 * @method jsGet_version
	 * @since 0.1.0
	 * @hidden
	 */
	open fun jsGet_version(callback: JavaScriptGetterCallback) {
		callback.returns(Build.VERSION.RELEASE)
	}
}
